### PR TITLE
BaSP-TG-20: Employee Tests

### PR DESCRIPTION
**Pull request by TomasBettini**

### CODE REVIEW

File: src/tests/employees.test.js [hunk 1]
```diff
@@ -138,4 +138,25 @@ describe('Time-sheet - Unit tests', () => {
...
     });
   });
+
+  describe('DELETE /:id', () => {
+    test('Should return status code 200', async () => {
+      const response = await request(app).delete('/employees/634d5803354e41cd60b9e401').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      console.log(response.body.message);
```
Comment (mara666) [inline src/tests/employees.test.js]: Eliminar console.log()

**Review by kolodani — COMMENTED** [review]
wonderful work

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "app_name is a system that provides Human Resources services to companies.",
   "main": "index.js",
   "scripts": {
-    "test": "jest",
+    "test": "jest --verbose",
```
Comment (lmalvasia) [inline package.json:7]: Is this necessary? Why did you add this flag?

File: src/models/Employees.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ const { Schema } = mongoose;
 const employeesSchema = new Schema({
   name: { type: String, required: true },
   lastName: { type: String, required: true },
-  phone: { type: Number, required: true },
+  phone: { type: String, required: true },
```
Comment (lmalvasia) [inline src/models/Employees.js:8]: Good job 😂

File: src/tests/employees.test.js [hunk 4]
```diff
@@ -1,5 +1,280 @@
...
+    test('Should return status code 200', async () => {
+      const response = await request(app).get('/employees/634d5803354e41cd60b9e400').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Employee found');
+    });
+  });
+
+  describe('GET BY ID ERROR /employees/:id', () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:31]: It's not necessary to have a different describe to separate the errors and the success responses.

File: src/tests/employees.test.js [hunk 5]
```diff
@@ -1,5 +1,280 @@
...
+    test('Should return status code 404', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:32]: Why the request is returning a status 404?
You have to add the purpose why this is going to return an error. For example:
```suggestion
    test('should return an status 400 when the employee does not exist in the database', async () => {
```

File: src/tests/employees.test.js [hunk 6]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).post('/employees').send(employeeMocked);
+      expect(response.status).toBe(201);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Employee created successfully');
+    });
+  });
+
+  describe('POST ERROR Name Required /employees', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:58]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 7]
```diff
@@ -1,5 +1,280 @@
...
+      };
+      const response = await request(app).post('/employees').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "name" is not allowed to be empty ');
+    });
+  });
+
+  describe('POST ERROR Last Name Required /employees', () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:74]: This is not necessary, you have lot of describes. Please, move all of these tests inside only one describe.

File: src/tests/employees.test.js [hunk 8]
```diff
@@ -1,5 +1,280 @@
...
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:75]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 9]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).post('/employees').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "lastName" is not allowed to be empty ');
+    });
+  });
+
+  describe('POST ERROR Phone Required /employees', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:92]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 10]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).post('/employees').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "phone" is not allowed to be empty ');
+    });
+  });
+
+  describe('POST ERROR Email Required /employees', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:109]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 11]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).post('/employees').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "email" is not allowed to be empty ');
+    });
+  });
+
+  describe('POST ERROR Password Required /employees', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:126]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 12]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).delete('/employees/634d5803354e41cd60b9e401').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Employee with id: 634d5803354e41cd60b9e401 deleted');
+    });
+  });
+
+  describe('DELETE ERROR /employees/:id', () => {
+    test('Should return status code 404', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:153]: Why the request is returning a status 404?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 13]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).put('/employees/634d5803354e41cd60b9e405').send(employeeMocked);
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Employee id: 634d5803354e41cd60b9e405 edited');
+    });
+  });
+
+  describe('PUT ERROR ID /employees/:id', () => {
+    test('Should return status code 404', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:180]: Why the request is returning a status 404?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 14]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).put('/employees/634d5803354e41cd60b9e450').send(employeeMocked);
+      expect(response.status).toBe(404);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Employee id: 634d5803354e41cd60b9e450 does not exists');
+    });
+  });
+
+  describe('PUT ERROR Name Required /employees/:id', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:197]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 15]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).put('/employees/634d5803354e41cd60b9e409').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "name" is not allowed to be empty ');
+    });
+  });
+
+  describe('PUT ERROR Last Name Required /employees/:id', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:214]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 16]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).put('/employees/634d5803354e41cd60b9e409').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "lastName" is not allowed to be empty ');
+    });
+  });
+
+  describe('PUT ERROR Phone Required /employees/:id', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:231]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 17]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).put('/employees/634d5803354e41cd60b9e409').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "phone" is not allowed to be empty ');
+    });
+  });
+
+  describe('PUT ERROR Email Required /employees/:id', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:248]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/employees.test.js [hunk 18]
```diff
@@ -1,5 +1,280 @@
...
+      const response = await request(app).put('/employees/634d5803354e41cd60b9e409').send(employeeMocked);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe('Something went wrong: "email" is not allowed to be empty ');
+    });
+  });
+
+  describe('PUT ERROR Password Required /employees/:id', () => {
+    test('Should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/employees.test.js:265]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: package.json (hunk 2 above)
Comment (TomasBettini) [reply package.json:7]: Because with this flag we can see in more detail which are the tests that fail or pass, but if you want I can delete it

**Review by lmalvasia — APPROVED** [review]
